src/main/java/org/dependencytrack/model/Component.java

```diff
@@ -367,12 +370,13 @@ public enum FetchGroup {
     private List<ComponentProperty> properties;
 
     @Persistent(table = "COMPONENTS_VULNERABILITIES")
-    @Join(column = "COMPONENT_ID")
-    @Element(column = "VULNERABILITY_ID")
+    @Join(column = "COMPONENT_ID", foreignKey = "COMPONENTS_VULNERABILITIES_COMPONENT_FK", deleteAction = ForeignKeyAction.CASCADE)
+    @Element(column = "VULNERABILITY_ID", foreignKey = "COMPONENTS_VULNERABILITIES_VULNERABILITY_FK", deleteAction = ForeignKeyAction.CASCADE)
```

Comment:
To verify that this syntax does what we want, I let DataNucleus generate the schema based on the annotations alone (i.e., what we did in DT v4). It generated this:

```sql
ALTER TABLE "COMPONENTS_VULNERABILITIES" ADD CONSTRAINT "COMPONENTS_VULNERABILITIES_VULNERABILITY_FK" FOREIGN KEY ("VULNERABILITY_ID") REFERENCES "VULNERABILITY" ("ID") ON DELETE CASCADE INITIALLY DEFERRED ;

ALTER TABLE "COMPONENTS_VULNERABILITIES" ADD CONSTRAINT "COMPONENTS_VULNERABILITIES_COMPONENT_FK" FOREIGN KEY ("COMPONENT_ID") REFERENCES "COMPONENT" ("ID") ON DELETE CASCADE INITIALLY DEFERRED ;
```

Which matches what we defined in the DB changelog: https://github.com/DependencyTrack/hyades-apiserver/blob/ac840678ac3e98b06b2449ca08db9ec9c18727b7/src/main/resources/migration/changelog-v5.6.0.xml#L245-L255